web/src/hooks/viewport.js

```diff
@@ -1,49 +1,35 @@
-import { useState, useEffect } from 'react';
+import { useState, useEffect, useCallback } from 'react';
 
-export function useWidthObserver(ref, offset = 0) {
+export function useRefWidthHeightObserver(offsetX = 0, offsetY = 0) {
   const [width, setWidth] = useState(0);
+  const [height, setHeight] = useState(0);
+  const [node, setNode] = useState(null); // The DOM node
 
-  // Resize hook
-  useEffect(() => {
-    const updateWidth = () => {
-      if (ref.current) {
-        const newWidth = ref.current.getBoundingClientRect().width - offset;
-        if (newWidth !== width) { setWidth(newWidth); }
+  // See https://reactjs.org/docs/hooks-faq.html#how-can-i-measure-a-dom-node
+  const ref = useCallback((newNode) => {
+    // This callback will be called once the ref
+    // returned has been attached to `node`.
+    const update = () => {
+      if (newNode !== null) {
+        const newWidth = newNode.getBoundingClientRect().width - offsetX;
+        const newHeight = newNode.getBoundingClientRect().height - offsetY;
+        if (width !== newWidth) { setWidth(newWidth); }
+        if (height !== newHeight) { setHeight(newHeight); }
+        if (node !== newNode) { setNode(newNode); }
       }
     };
     // Set initial width
-    updateWidth();
+    update();
     // Update container width on every resize
-    window.addEventListener('resize', updateWidth);
-    return () => {
-      window.removeEventListener('resize', updateWidth);
-    };
-  });
```

Comment:
I really don't know what was going through my mind when I didn't list the dependencies here :confused: I'm glad you caught this! :+1: